pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/argoproj.io/v1alpha1/Workflow/customizations.yaml

```diff
@@ -0,0 +1,449 @@
+kind: ResourceInterpreterCustomization
+metadata:
+  name: declarative-configuration-workflow
+spec:
+  target:
+    apiVersion: argoproj.io/v1alpha1
+    kind: Workflow
+  customizations:
+    replicaResource:
+      luaScript: >
+        local kube = require("kube")
+        function GetReplicas(obj)
+          replica = 1
+          if obj.spec.parallelism ~= nil then
+            replica = obj.spec.parallelism
+          end
+          podTemplateSpec = {}
+          podTemplateSpec.spec = {}
+          podTemplateSpec.spec.affinity = obj.spec.affinity
+          podTemplateSpec.spec.nodeSelector = obj.spec.nodeSelector
+          podTemplateSpec.spec.tolerations = obj.spec.tolerations
+          requirement = kube.accuratePodRequirements(podTemplateSpec)
+          return replica, requirement
+        end
+    replicaRevision:
+      luaScript: >
+        function ReviseReplica(obj, replica)
+          obj.spec.parallelism = replica
+          return obj
+        end
+    healthInterpretation:
+      luaScript: >
+        function InterpretHealth(observedObj)
+          if observedObj.status.phase == nil or observedObj.status.phase == '' or observedObj.status.phase == 'Failed' or observedObj.status.failed == 'Error' then
+            return false
+          end
+          return true
+        end
+    statusAggregation:
+      luaScript: >
+        function AggregateStatus(desiredObj, statusItems)
+          if statusItems == nil then
+            return desiredObj
+          end
+          if desiredObj.status == nil then
+            desiredObj.status = {}
+          end
+          if desiredObj.status.conditions == nil then
+            desiredObj.status.conditions = {}
+          end       
+          phase = ''
+          startedAt = {}
+          finishedAt = {}
+          estimatedDuration = 0
+          progress = ''
+          message = ''
+          compressedNodes = ''
+          nodes = {}
+          offloadNodeStatusVersion = ''
+          storedTemplates = {}
+          persistentVolumeClaims = {} 
+          outputs = {}
+          storedWorkflowTemplateSpec = {}
+          synchronization = {}
+          artifactRepositoryRef = {}
+          artifactGCStatus = {}
+          generation = 0
+
+          resourcesDuration = {}
+          conditions = {}
+          type = ''
+          conditionsIndex = 0
+          workflowCompleted = {}
+          workflowPodRunning = {}
+          workflowSpecWarning = {}
+          workflowSpecError = {}
+          workflowMetricsError = {}
+          workflowArtifactGCError = {}
+          for i = 1, #statusItems do
+            if statusItems[i].status ~= nil and statusItems[i].status.phase ~= nil then
+              if phase == '' then
+                phase = phase..statusItems[i].clusterName..'='..statusItems[i].status.phase
+              else
+                phase = phase..','..statusItems[i].clusterName..'='..statusItems[i].status.phase
+              end
+            end
```

Comment:
Phase a simple, high-level summary of where the workflow is in its lifecycle. It will be "" (Unknown), "Pending", or "Running" before the workflow is completed, and "Succeeded", "Failed" or "Error" once the workflow has completed.[ref](https://github.com/argoproj/argo-workflows/blob/master/pkg/apis/workflow/v1alpha1/workflow_types.go#L1820)